x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.start_data_frame_transform.json

```diff
@@ -11,6 +11,13 @@
           "required": true,
           "description": "The id of the transform to start"
         }
+      },
+      "params": {
+        "timeout": {
+          "type": "time",
+          "required": false,
+          "description": "Controls the time to wait for the transform to start"
```

Comment:
would be nice to know the default value.

Reply:
I'm having second thoughts about this timeout parameter. Get Stats, Preview and Put could all take a timeout parameter as ActionRequests or BaseTasksRequests. Should we actually expose it? Is it helpful to offer this option or confusing? I think it makes sense for Stop and Start (start datafeed and open job both have a timeout parameter). I don't know anymore I'm over-thinking it.

Reply:
`GET stats` `_preview` and `PUT` should not really have a timeout option (IMO). 

`_start` could be take a long time as it needs to wait for a Node to start executing the task, thus a timeout sort of makes sense.

Addendum, if we include `timeout` we may also want to include a `wait_for_completion` or a `sync/async` option, but that could be another PR and discussion.

Reply:
I agree that `_start` and `_stop` should take a timeout parameter and the others shouldn't.  For `_start` and `_stop` there will be occasions when the caller wants to know when the state change is complete before they move on to the next operation, yet they probably don't want to wait forever if something goes wrong.  Our integration tests are a good example of this need but any user trying to automate an end-to-end process that involves data frames will be in a similar situation.  Having a timeout on the `_start` and `_stop` endpoints is also consistent with what we do for datafeeds and anomaly detectors.

Reply:
Makes sense. Thanks for the 2nd review @benwtrent